stdlib/Random/test/runtests.jl

```diff
@@ -1250,6 +1250,12 @@ end
     @test length(xs) == 3
 end
 
+@testset "Float32 RNG typo" begin
+    for T in (Float16, Float32, Float64)
+        @test length(unique(rand(T, 14))) > 10
```

Comment:
As mentioned at https://github.com/JuliaLang/julia/pull/57338#discussion_r1948993368, this is testing only one codepath for generating random numbers, it'd be better to test both
```suggestion
        # Make sure generated numbers are sufficiently diverse
        # for both SIMD and non-SIMD RNG code paths.
        @test length(unique!(rand(T, 14))) > 10
        @test length(unique!(rand(T, 34))) > 20
```